crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -77,6 +77,34 @@ impl Auction {
                             competition::order::Kind::Limit { .. } => Class::Limit,
                             competition::order::Kind::Liquidity => Class::Liquidity,
                         },
+                        protocol_fees: order
```

Comment:
How are we treating protocol fees specified in the app data here?
Those will not be part of this array, right?

Reply:
I don't see a good reason to complicate things and treat partner fee differently from other types of fees solvers (and eventually drivers) need to withhold. So either:
1. Driver withholds all type of fees or
2. All types of fees are forwarded to solvers and driver doesn't do anything.

Reply:
I don't really understand the response.
Intuitively when I see this API I would assume that for example if the array is empty I'm not supposed to take any fee for the order. But there is another PR open that parses a partner fee from the app data. So would solvers be expected to know that they have to parse the appdata as well to learn about a potential partner fee?

Reply:
I agree. We should define a per-solver flag which tells us if they want to handle the fees themselves. If this flag is true they
1. Receive all fee policies
2. The driver doesn't adjust any prices

Otherwise they don't receive any fee policies and the driver adjusts all prices.

Reply:
Implemented as suggested @fleupold